### PR TITLE
WIP: Feature/unique dev label

### LINKED ISSUE #308: Make dev label a unique string
Clusters need to have a unique dev label
Meaning:
dev=foo
dev=bar

The controller should get started with:
```bash
kubermatic-controller-manager --dev=foo
```

The controller should now only work on those clusters which have the label:
```yaml
dev=foo
```

### PR DESCRIPTION
**Pull request by lbb**
Resolves #308

### CODE REVIEW

**Review by scruplelesswizard — APPROVED** [review]

File: controller/cluster/controller_test.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ func newTestController() (*fake.Clientset, *clusterController) {
 	updates := buildMasterUpdates()
 
 	clientSet := fake.NewSimpleClientset()
-	cc, err := NewController("", clientSet, tprClient, etcdClient, cps, versions, updates, "./../../master-resources/", "localhost", true, "./../../addon-charts/")
+	cc, err := NewController("", clientSet, tprClient, etcdClient, cps, versions, updates, "./../../master-resources/", "localhost", "true", "./../../addon-charts/")
```
Comment (mrIncompetent) [inline controller/cluster/controller_test.go]: Replace with a proper key

File: handler/routing_test.go [hunk 2]
```diff
@@ -34,7 +34,7 @@ func createTestEndpoint() http.Handler {
 	// create CloudProviders
 	cps := cloud.Providers(dcs)
 
-	kps, err := kubernetes.Providers("./fixtures/kubecfg.yaml", dcs, cps, "./fixtures/secrets.yaml", true)
+	kps, err := kubernetes.Providers("./fixtures/kubecfg.yaml", dcs, cps, "./fixtures/secrets.yaml", "true")
```
Comment (mrIncompetent) [inline handler/routing_test.go]: Replace with a proper key

**Review by mrIncompetent — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -48,7 +48,7 @@ Example for `kubermatic-api`
 
 ```
 make build CMD=kubermatic-api && ./kubermatic-api \
---dev=1 \
+--dev="unique-label-abcdef123" \
```
Comment (mrIncompetent) [inline README.md]: Outdated

File: README.md [hunk 4]
```diff
@@ -64,14 +64,17 @@ make build CMD=kubermatic-cluster-controller &&  \
 ./kubermatic-cluster-controller \
 --datacenters=$GOPATH/src/github.com/kubermatic/config/seed-clusters/dev.kubermatic.io/datacenters.yaml \
 --kubeconfig=$GOPATH/src/github.com/kubermatic/config/seed-clusters/dev.kubermatic.io/kubeconfig \
---dev=1 \
+--dev="unique-label-abcdef123" \
```
Comment (mrIncompetent) [inline README.md]: Outdated

File: README.md [hunk 5]
```diff
@@ -64,14 +64,17 @@ make build CMD=kubermatic-cluster-controller &&  \
...
 --logtostderr=1 \
 --master-resources=$GOPATH/src/github.com/kubermatic/config/kubermatic/static/master \
 --v=4 \
 --addon-resources=$GOPATH/src/github.com/kubermatic/api/addon-charts \
 --external-url=dev.kubermatic.io
 ```
 
+Valid dev label value must be 63 characters or less and must be empty or begin and end with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_), dots (.), and alphanumerics between.
```
Comment (mrIncompetent) [inline README.md]: Outdated

File: provider/kubernetes/kubernetes.go [hunk 6]
```diff
@@ -35,17 +35,17 @@ type kubernetesProvider struct {
...
+	workerName string
+	config     *rest.Config
 }
 
 // NewKubernetesProvider creates a new kubernetes provider object
 func NewKubernetesProvider(
 	clientConfig *rest.Config,
 	cps map[string]provider.CloudProvider,
-	dev bool,
+	dev string,
```
Comment (mrIncompetent) [inline provider/kubernetes/kubernetes.go]: Outdated

File: provider/kubernetes/kubernetes.go [hunk 7]
```diff
@@ -62,7 +62,7 @@ func NewKubernetesProvider(
 		cps:              cps,
 		kuberntesClient:  client,
 		tprClient:        trpClient,
-		dev:              dev,
+		workerName:       dev,
```
Comment (mrIncompetent) [inline provider/kubernetes/kubernetes.go]: Outdated

**Review by mrIncompetent — CHANGES_REQUESTED** [review]

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**